cylc/flow/scheduler.py

```diff
@@ -1050,6 +1049,133 @@ def kill_tasks(
 
         return len(unkillable)
 
+    def remove_tasks(
+        self, items: Iterable[str], flow_nums: Optional['FlowNums'] = None
+    ) -> None:
+        """Remove tasks from the task pool (by command).
+
+        Args:
+            items: Relative IDs or globs.
+            flow_nums: Flows to remove the tasks from. If empty or None, it
+                means 'all'.
+        """
+        active, inactive, _unmatched = self.pool.filter_task_proxies(
+            items, warn_no_active=False, inactive=True
+        )
+        if not (active or inactive):
+            return
+
+        if flow_nums is None:
+            flow_nums = set()
+        # Mapping of task IDs to removed flow numbers:
+        removed: Dict[str, FlowNums] = {}
+        not_removed: Set[str] = set()
+        to_kill: List[TaskProxy] = []
+
+        for itask in active:
+            fnums_to_remove = itask.match_flows(flow_nums)
+            if not fnums_to_remove:
+                not_removed.add(itask.identity)
+                continue
+            removed[itask.identity] = fnums_to_remove
+            if fnums_to_remove == itask.flow_nums:
+                # Need to remove the task from the pool.
+                # Spawn next occurrence of xtrigger sequential task (otherwise
+                # this would not happen after removing this occurrence):
+                self.pool.check_spawn_psx_task(itask)
+                self.pool.remove(itask, 'request')
+                to_kill.append(itask)
+                itask.removed = True
+            itask.flow_nums.difference_update(fnums_to_remove)
+
+        matched_task_ids = {
+            *removed.keys(),
+            *(quick_relative_id(cycle, task) for task, cycle in inactive),
+        }
+
+        for id_ in matched_task_ids:
+            point_str, name = id_.split('/', 1)
+            tdef = self.config.taskdefs[name]
+            # Go through downstream tasks to see if any need to stand down
+            # as a result of this task being removed:
```

Comment:
There are 2 things happening in this loop:
1. Go through downstream tasks to see if any need to stand down
2. Remove the matched task in the DB

Both of these comments come right before what they say they are doing.

As for both removed active tasks and matched inactive ones... the removed active tasks are matched by definition, no?

Reply:
Fair enough. I guess I was thinking of "stand down", which is not exactly standard terminology for us, as covering both 1. and 2.  

> As for both removed active tasks and matched inactive ones... the removed active tasks are matched by definition, no?

The problem I was trying to address is that I didn't initially see that this handles both active and inactive tasks, because the first loop handles only active tasks. 

I was probably just being dense and making assumptions based on the fact that we quite a few methods that handle active and inactive tasks separately one after the other, but maybe you could just add a slightly more comprehensive comment or two yourself? No biggie though.

Reply:
Rewritten somewhat, continue this discussion in https://github.com/MetRonnie/cylc-flow/pull/34